### PR TITLE
feat(recurringjob): add RecurringJobMaxRetention setting

### PR DESCRIPTION
**Pull request by mantissahz**
Add the `recurringJobMaxRetention` default setting and update the README.md file.

ref: longhorn/longhorn#5713

### CODE REVIEW

**Comment by innobead** [comment]
@jillian-maroket Please help review the description of the newly added setting.

File: chart/README.md [hunk 1]
```diff
@@ -305,6 +305,7 @@ For more details like types or options, you can refer to **Settings Reference**
 | defaultSettings.orphanAutoDeletion | This setting allows Longhorn to delete the orphan resource and its corresponding orphaned data automatically like stale replicas. Orphan resources on down or unknown nodes will not be cleaned up automatically. |
 | defaultSettings.priorityClass | priorityClass for Longhorn system-managed components This setting can help prevent Longhorn components from being evicted under Node Pressure. Notice that this will be applied to Longhorn user-deployed components by default if there are no priority class values set yet, such as `longhornManager.priorityClass`. |
 | defaultSettings.recurringFailedJobsHistoryLimit | This setting specifies how many failed backup or snapshot job histories should be retained. History will not be retained if the value is 0. |
+| defaultSettings.recurringJobMaxRetention | This setting specifies how many snapshots or backups should be retained |
```
Comment (jillian-maroket) [inline chart/README.md:308]:
```suggestion
| defaultSettings.recurringJobMaxRetention | Maximum number of snapshots or backups to be retained. |
```

File: chart/questions.yaml [hunk 2]
```diff
@@ -374,6 +374,12 @@ questions:
     type: int
     min: 0
     default: 1
+  - variable: defaultSettings.recurringJobMaxRetention
+    label: Maximum Retention Number for Recurring Job
+    description: "This setting specifies how many snapshots or backups should be retained."
```
Comment (jillian-maroket) [inline chart/questions.yaml:379]:
```suggestion
    description: "Maximum number of snapshots or backups to be retained."
```

File: chart/values.yaml [hunk 3]
```diff
@@ -222,6 +222,8 @@ defaultSettings:
   recurringSuccessfulJobsHistoryLimit: ~
   # -- This setting specifies how many failed backup or snapshot job histories should be retained. History will not be retained if the value is 0.
   recurringFailedJobsHistoryLimit: ~
+  # -- This setting specifies how many snapshots or backups should be retained
```
Comment (jillian-maroket) [inline chart/values.yaml:225]:
```suggestion
  # -- Maximum number of snapshots or backups to be retained.
```

**Review by jillian-maroket — COMMENTED** [review]
Let me know if you have concerns about the changes.

File: chart/README.md (hunk 1 above)
Comment (mantissahz) [reply chart/README.md]: Modified. Thanks

File: chart/questions.yaml (hunk 2 above)
Comment (mantissahz) [reply chart/questions.yaml]: Modified. Thanks

File: chart/values.yaml (hunk 3 above)
Comment (mantissahz) [reply chart/values.yaml]: Modified. Thanks

**Review by jillian-maroket — APPROVED** [review]

**Review by innobead — APPROVED** [review]
LGTM.

**Comment by innobead** [comment]
@mantissahz Please help update longhorn/longhorn to make the description sync.

**Pull request merged**